test/parallel/test-stream-finished.js

```diff
@@ -273,27 +273,26 @@ const { promisify } = require('util');
 }
 
 {
-  // Premature close if stream never emitted 'finish'
-  // even if writableFinished says something else.
+  // No error if stream never emitted 'end'
+  // even if readableEnded says something else.
 
   const streamLike = new EE();
   streamLike.writable = true;
-  finished(streamLike, common.expectsError({
-    code: 'ERR_STREAM_PREMATURE_CLOSE'
+  finished(streamLike, common.mustCall((err) => {
+    assert.strictEqual(err, undefined);
   }));
```

Comment:
Would you mind explaining the reason behind this? Thanks. Was this introduced in another PR?

Reply:
This test was introduced in another PR (the one that caused the referenced issued). In this PR I've loosened the restriction to e.g. requiring the stream to be `readableEnded` instead of having actually emitted `'end'`. If you have a stream that thinks it is done it shouldn't matter if it actually have emitted `'end'`. Though the ordering of `'close'` and `'end'` is no longer enforced in this specific scenario.

I don't think this is required to fix the referenced issue but I'm hoping it leaves a bit more wiggle room for compat for not strictly "correct" streams.

Would you prefer I try to keep this PR to minimal change?